test/e2e/create_test.go

```diff
@@ -53,18 +63,33 @@ func testWindowsNodeCreation(t *testing.T) {
 			Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval}); err != nil {
 		t.Fatalf("error creating wcmo custom resource  %v", err)
 	}
-	// As per testing, each windows VM is taking roughly 12 minutes to be shown up in the cluster, so to be on safe
-	// side, let's make it as 60 minutes. The value comes from timeout variable
 	err = testCtx.waitForWindowsNode()
 	if err != nil {
 		t.Fatalf("windows node creation failed  with %v", err)
 	}
 }
 
+// testNodeCreation tests if the expected number of Windows nodes were created
+func testNodeCreation(t *testing.T) {
```

Comment:
Why to create a separate test for this? Why can't we populate it in the `waitForWindowsNode` method? Are you concerned about staleness?

Reply:
Sure I can do that. I also see `nodes` in the `testContext`. This is going to get really convoluted pretty quickly. We should have this only in one spot.

Reply:
Whatever we have in global context shouldn't be local context unless, it has staleness concerns.

Reply:
>Why can't we populate it in the waitForWindowsNode method? Are you concerned about staleness?

Turns out that if I initialize `gc.nodes` here, hybrid overlay MAC annotations are not present yet. The only option I see is to either make it wait here for the annotations here or to initialize `gc.nodes` later like I am doing.

Reply:
Or reinitialize `gc.nodes` in the network test case